nomad/leader.go

```diff
@@ -531,6 +531,7 @@ func (s *Server) reapFailedEvaluations(stopCh chan struct{}) {
 			followupEvalWait := s.config.EvalFailedFollowupBaselineDelay +
 				time.Duration(rand.Int63n(int64(s.config.EvalFailedFollowupDelayRange)))
 			followupEval := eval.CreateFailedFollowUpEval(followupEvalWait)
+			updateEval.NextEval = followupEval.ID
```

Comment:
Can we add a comment to CreateFailedFollowUpEval that callers should copy the new eval ID to the old eval?

We should also update the comment on Prev/NextEval on the Evaluation struct as they only mention deployments.